.github/workflows/test.yml

```diff
@@ -15,18 +13,21 @@ jobs:
       fail-fast: false
       matrix:
         os:
-        - ubuntu-latest
-        - macOS-latest
-        - windows-latest
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
     steps:
-    - name: checkout
-      uses: actions/checkout@v4
+      - name: checkout
+        uses: actions/checkout@v4
 
-    - name: setup go
-      uses: actions/setup-go@v5
-      with:
-        go-version-file: go.mod
+      - name: setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
 
-    - name: test
-      run: go test ./...
+      - name: test
+        run: make test
 
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        if: matrix.os == 'ubuntu-latest'
```

Comment:
Does this require some credentials? How does codecov work?

Reply:
No credentials required for open source